libs/sdk-ui-ext/src/dashboardView/DashboardRenderer.tsx

```diff
@@ -106,6 +108,43 @@ const polluteWidgetRefsWithBothIdAndUri = (
         return updatedContent;
     });
 
+const validateItemsSize = (
+    getInsightByRef: (insightRef: ObjRef) => IInsight | undefined,
+    enableKDWidgetCustomHeight: boolean,
+): DashboardLayoutItemModifications => (item) => {
+    const widget = item.facade().widget();
+    if (isInsightWidget(widget)) {
```

Comment:
Should we throw else? If I understand correctly, this if is for type narrowing and we do not expect this to be called with non insight right?

Reply:
Yes, basically no other widget type then insight, can currently change its inner content affecting valid size. But not sure if some error is needed, other types should just be ignored and untouched